### PR TITLE
8282106: [lworld] IdentityException should identify the value class

### PR DESCRIPTION
**Pull request by fparain**
Improve the message of IdentityException by including the class of the instance that triggered the exception.

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Issue
 * [JDK-8282106](https://bugs.openjdk.org/browse/JDK-8282106): [lworld] IdentityException should identify the value class (**Bug** - P4)

### Reviewers
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/valhalla.git pull/1150/head:pull/1150` \
`$ git checkout pull/1150`

Update a local copy of the PR: \
`$ git checkout pull/1150` \
`$ git pull https://git.openjdk.org/valhalla.git pull/1150/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1150`

View PR using the GUI difftool: \
`$ git pr show -t 1150`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/valhalla/pull/1150.diff">https://git.openjdk.org/valhalla/pull/1150.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/valhalla/pull/1150#issuecomment-2195661554)

### CODE REVIEW

File: test/hotspot/jtreg/runtime/valhalla/inlinetypes/MonitorEnterTest.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+*/
+
+package runtime.valhalla.inlinetypes;
+
+import jdk.test.lib.Asserts;
+
+public class MonitorEnterTest {
+
+  static void monitorEnter(Object o, boolean expectSuccess, String message) {
+    try {
```
Comment (TobiHartmann) [inline test/hotspot/jtreg/runtime/valhalla/inlinetypes/MonitorEnterTest.java:55]: Should be 4-whitespace indentation for Java code.

**Review by TobiHartmann — COMMENTED** [review]
What about AArch64? Wouldn't the test now fail on that platform?

**Comment by fparain** [comment]
Thank your for the reminder for aarch64 (the Mach5 job I ran last night reminded it to me as well).
Now aarch64 changes are included (they are similar to the x64 changes).
Failed tests on Mach5 have been tested locally, and there's a new Mach5 job testing the updated patch.

File: test/hotspot/jtreg/runtime/valhalla/inlinetypes/MonitorEnterTest.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+* or visit www.oracle.com if you need additional information or have any
+* questions.
+*/
+
+/**
+* @test TestCloneableValue
+* @library /test/lib
+* @enablePreview
+* @compile TestCloneableValue.java
+* @run main/othervm -XX:LockingMode=0 runtime.valhalla.inlinetypes.TestCloneableValue
```
Comment (RogerRiggs) [inline test/hotspot/jtreg/runtime/valhalla/inlinetypes/MonitorEnterTest.java:29]: I'm missing the connection between TestCloneableValue and MonitorEnterTest?
Should the \@run line refer to MonitorEnterTest.MyValue?

Comment (fparain) [reply test/hotspot/jtreg/runtime/valhalla/inlinetypes/MonitorEnterTest.java]: Good catch! This is not the test I used during development, but one that I wrote after, and I clearly did it too quickly.
I've pushed a fix for the test that should address this issue (and a few others in the test itself).

**Review by RogerRiggs — COMMENTED** [review]
The test looks ok to me.

**Review by TobiHartmann — APPROVED** [review]
Looks good to me.

**Comment by fparain** [comment]
Thank you for the reviews.

**Pull request closed without merging**